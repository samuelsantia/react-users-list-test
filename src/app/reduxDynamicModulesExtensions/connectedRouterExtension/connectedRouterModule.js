import { connectRouter } from 'connected-react-router';

/**
 * redux-dynamic-moduler router module
 * loads the router reducer
 * @param {History} history router history
 * @return {Object} router module config
 */
export const getRouterModule = (history) => ({
  id: 'router',
  reducerMap: {
    router: connectRouter(history),
  },
});
