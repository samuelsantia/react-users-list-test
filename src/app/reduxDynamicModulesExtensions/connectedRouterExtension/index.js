export { getConnectedRouterExtension } from './connectedRouterExtension';
export { getConnectedRouterModule } from './connectedRouterModule';
