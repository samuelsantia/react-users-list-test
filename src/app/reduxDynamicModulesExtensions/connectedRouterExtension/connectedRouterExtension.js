import { routerMiddleware } from 'connected-react-router';
import { getRouterModule } from './connectedRouterModule';

/**
 * creates redux-dynamic-modules router extension
 * that register the middleware and load the router module
 * @param {History} history router history
 * @return {Object} extension config
 */
export const getConnectedRouterExtension = (history) => ({
  // adds router middleware to redux-dynamic-modules store
  middleware: [routerMiddleware(history)],
  onModuleManagerCreated: (moduleManager) => {
    // adds router module to redux-dynamic-modules
    moduleManager.addModule(getRouterModule(history));
  },
});
