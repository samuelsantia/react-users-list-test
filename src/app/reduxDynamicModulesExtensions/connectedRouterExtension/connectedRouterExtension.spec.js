import { routerMiddleware } from 'connected-react-router';
import { getRouterModule } from './connectedRouterModule';
import { getConnectedRouterExtension } from './connectedRouterExtension';

jest.mock('connected-react-router');
jest.mock('./connectedRouterModule');

const fakeRouterMiddleware = (value) => value;
const fakeHistory = {};
const fakeRouterModule = {};

beforeEach(() => {
  routerMiddleware.mockReturnValue(fakeRouterMiddleware);
  getRouterModule.mockReturnValue(fakeRouterModule);
});

afterEach(() => {
  jest.resetAllMocks();
});

describe('getConnectedRouterExtension', () => {
  it('should return the routerMiddleware in middleware option', () => {
    const actual = getConnectedRouterExtension(fakeHistory).middleware;
    const expected = [fakeRouterMiddleware];

    expect(routerMiddleware).toHaveBeenCalledTimes(1);
    expect(routerMiddleware).toHaveBeenCalledWith(fakeHistory);
    expect(actual).toEqual(expected);
  });

  it('should add router module onModuleManagerCreated', () => {
    const addModuleMock = jest.fn();
    const moduleManager = { addModule: addModuleMock };

    const { onModuleManagerCreated } = getConnectedRouterExtension(fakeHistory);
    onModuleManagerCreated(moduleManager);

    expect(getRouterModule).toHaveBeenCalledTimes(1);
    expect(getRouterModule).toHaveBeenCalledWith(fakeHistory);

    expect(addModuleMock).toHaveBeenCalledTimes(1);
    expect(addModuleMock).toHaveBeenCalledWith(getRouterModule(fakeHistory));
  });

  it('should match snapshot', () => {
    expect(getConnectedRouterExtension(fakeHistory)).toMatchSnapshot();
  });
});
