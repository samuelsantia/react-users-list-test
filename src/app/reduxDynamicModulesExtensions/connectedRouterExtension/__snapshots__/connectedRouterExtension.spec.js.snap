// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getConnectedRouterExtension should match snapshot 1`] = `
Object {
  "middleware": Array [
    [Function],
  ],
  "onModuleManagerCreated": [Function],
}
`;
